cmake/patches/fp16/remove_math_h_dependency_from_fp16_h.patch

```diff
@@ -0,0 +1,29 @@
+diff --git a/include/fp16/fp16.h b/include/fp16/fp16.h
+index 2b61fff..1947805 100644
+--- a/include/fp16/fp16.h
++++ b/include/fp16/fp16.h
+@@ -4,10 +4,8 @@
+ 
+ #if defined(__cplusplus) && (__cplusplus >= 201103L)
+ 	#include <cstdint>
+-	#include <cmath>
```

Comment:
Do we want to do this for all builds, or limit the change to Apple + x86_64 [+ Xcode 16+]?

Reply:
the change seems like it should be fine for all builds, but I've limited it to Apple x86_64 targets for now.

the Xcode 16 constraint is a little trickier. a non-Xcode generator like "Unix Makefiles" may use the same compiler. I'm also not sure whether this is something we should expect to be fixed in a future Xcode version.